internal/venice-client-common/src/main/java/org/apache/avro/generic/DeterministicMapOrderDatumWriter.java

```diff
@@ -14,20 +16,46 @@
  * the same as natural ordering of map keys.
  */
 public interface DeterministicMapOrderDatumWriter {
```

Comment:
This is not just in the key schema. I believe it was originally added for AA collection merging. Do we need to put this support in fastavro, then?

Reply:
Yeah, I think it is good to support such behavior in fast-avro.

@FelixGV 
Do you recall the original reason of introducing this deterministic map serialization? I supposed it is about key schema usage and later on, it is being used by AA collection merging.

I think it should be fine to use this deterministic map serialization as the default writer for map type since the serialized bytes can always be deserialized by either fast-avro or vanilla avro.

Reply:
I believe it is just for map keys, yeah. Technically it adds unneeded overhead for the maps inside values (which should be much more common than map keys...).

Reply:
I see.
Maybe control it by some feature flag in fast-avro.

Reply:
I think we can document or cut a ticket to track the key schema issue.
If we make a change to use deterministic map serialization for key schema in clients, it may start breaking things since the write path may not be using it.

This remind me that we recently made a change to Samza Producer to adopt the fast-avro, which would change the behavior.

@sixpluszero 
I think we may need to rethink about the fast-avro adoption for key object serialization in VeniceSystemProducer.
Such as continuing to use the `AvroSerializer` when the key schema contains `map` structure.

Reply:
I think Alexei also recently merged a change to use fast-avro everywhere in the client... this might also need another look. We should write an integ test with a map key to ensure we don't break things...

Reply:
I created an internal ticket for the team to investigate this (Ticket 10714)